Fix ts24

tbosch opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

mary-poppins commented:
You can preview cd2ba3e at https://pr19097-cd2ba3e.ngbuilds.io/.

tbosch commented on npm-shrinkwrap.clean.json:
```diff
@@ -1,12 +1,18 @@
   "dependencies": {
     "@bazel/typescript": {
-      "version": "0.0.9",
+      "version": "0.0.7",
```
Why this downgrade?

tbosch replied:
This is in angular-srcs, why do we need this?

tbosch commented on npm-shrinkwrap.clean.json:
```diff
@@ -1,12 +1,18 @@
+          "version": "0.23.6"
+        },
+        "typescript": {
+          "version": "2.3.4"
```
This should be 2.4?

tbosch commented on npm-shrinkwrap.clean.json:
```diff
@@ -1,12 +1,18 @@
           "version": "7.0.18"
+        },
+        "tsickle": {
+          "version": "0.23.6"
```
This should be 0.24.0 as well (see below)

mary-poppins commented:
You can preview df17893 at https://pr19097-df17893.ngbuilds.io/.

mary-poppins commented:
You can preview bc8df03 at https://pr19097-bc8df03.ngbuilds.io/.

mary-poppins commented:
You can preview d7cd344 at https://pr19097-d7cd344.ngbuilds.io/.

mary-poppins commented:
You can preview 4a606f3 at https://pr19097-4a606f3.ngbuilds.io/.

mary-poppins commented:
You can preview 8c7a3d3 at https://pr19097-8c7a3d3.ngbuilds.io/.

mary-poppins commented:
You can preview 71f2297 at https://pr19097-71f2297.ngbuilds.io/.

mary-poppins commented:
You can preview 45b041f at https://pr19097-45b041f.ngbuilds.io/.

mary-poppins commented:
You can preview 26702fd at https://pr19097-26702fd.ngbuilds.io/.

mary-poppins commented:
You can preview 4e458c3 at https://pr19097-4e458c3.ngbuilds.io/.

mary-poppins commented:
You can preview edb1d8e at https://pr19097-edb1d8e.ngbuilds.io/.

IgorMinar commented on scripts/ci/install.sh:
```diff
@@ -42,7 +42,7 @@ travisFoldEnd "npm-install"
 # Install Selenium WebDriver
 travisFoldStart "webdriver-manager-update"
-  $(npm bin)/webdriver-manager update
+  $(npm bin)/webdriver-manager update --gecko false
```
can you add a comment about the gecko flag?

tbosch replied:
Done

IgorMinar reviewed (approved):
otherwise lgtm

mary-poppins commented:
You can preview 39aebc3 at https://pr19097-39aebc3.ngbuilds.io/.

tbosch committed:
refactor: update angular to support TypeScript 2.4

Detailed updates:
- rxjs@5.0.x
- tsickle@0.24.x
- typescript@2.4.x
- @bazel/typescript@0.10.0
- protractor@5.1.x
- selenium-webdriver@3.0.x

BREAKING CHANGE:
- the Angular compiler now requires TypeScript 2.4.x.

mary-poppins commented:
You can preview 6799c7f at https://pr19097-6799c7f.ngbuilds.io/.

gkalpak commented on packages/examples/upgrade/static/ts/module.ts:
```diff
@@ -140,7 +140,7 @@ ng1AppModule.factory(
 // #docregion downgrade-ng2-heroes-service
 // Register an AngularJS service, whose value is the "downgraded" Angular injectable.
-ng1AppModule.factory('heroesService', downgradeInjectable(HeroesService));
+ng1AppModule.factory('heroesService', downgradeInjectable(HeroesService) as any);
```
Why was this necessary? `downgradeInjectable()` returns `Function` and `.factory`'s second argument is of type `Function`?
This shows up in the docs, so we should avoid `as any`.

@tbosch, where were you getting errors without `as any`?
